src/common/device/AndroidDevice.ts

```diff
@@ -199,5 +235,156 @@ export class AndroidDevice implements BaseDevice {
+        await AndroidUtils.executeAdbCommand('root', this.port, this.logger);
+
+        // For API 29 or higher there are a few more steps to be done before we can remount after rooting
+        if (Version.sameOrNewer(this.osVersion, Version.from('29')!)) {
+            const verificationIsAlreadyDisabled = (
+                await AndroidUtils.executeAdbCommand('shell avbctl get-verification', this.port, this.logger)
+            ).includes('disabled');
+
+            const verityIsAlreadyDisabled = (
+                await AndroidUtils.executeAdbCommand('shell avbctl get-verity', this.port, this.logger)
+            ).includes('disabled');
+
+            if (!verificationIsAlreadyDisabled || !verityIsAlreadyDisabled) {
+                CommonUtils.updateCliAction(messages.getMessage('disableAVBVerityStatus'));
+            }
+
+            if (!verificationIsAlreadyDisabled) {
+                // Disable Android Verified Boot
+                await AndroidUtils.executeAdbCommand('shell avbctl disable-verification', this.port, this.logger);
+            }
+
+            if (!verityIsAlreadyDisabled) {
+                // Disable Verity
+                await AndroidUtils.executeAdbCommand('disable-verity', this.port, this.logger);
+            }
+
+            // If AVB and Verity were not disabled already and we had to run
+            // commands to disable them, then reboot for changes to take effect.
+            if (!verificationIsAlreadyDisabled || !verityIsAlreadyDisabled) {
+                CommonUtils.updateCliAction(messages.getMessage('rebootChangesStatus'));
+
+                // Reboot for changes to take effect
+                await this.reboot(true);
+
+                // Root again
+                await AndroidUtils.executeAdbCommand('root', this.port, this.logger);
+            }
+        }
+
+        CommonUtils.updateCliAction(messages.getMessage('remountSystemWritableStatus'));
+        // Now we're ready to remount and truly have root & writable access to system
+        await AndroidUtils.executeAdbCommand('remount', this.port, this.logger);
+    }
+
+    /**
+     * Attempts to launch the emulator and returns the ADB port that the emulator was launched on.
+     *
+     * @param bootMode Optional enum indicating the boot mode. Defaults to Normal.
+     * @param coldBoot Optional boolean indicating whether we should perform a cold boot. Defaults to false.
+     * @param waitForBoot Optional boolean indicating whether it should wait for the device to finish booting up. Defaults to true.
+     * @returns The ADB port that the emulator was launched on.
+     */
+    private async startEmulator(bootMode = BootMode.normal, coldBoot = false, waitForBoot = true): Promise<void> {
+        const port = await AndroidUtils.emulatorHasPort(this.id, this.logger);
+        const resolvedPortNumber = port ? port : await AndroidUtils.getNextAvailableAdbPort(this.logger);
+
+        if (resolvedPortNumber === port) {
```

Comment:
Nit: I would suggest pulling this boolean comparison out to a more meaningful name for this context. E.g.:

```typescript
const emulatorAlreadyBooted = (resolvedPortNumber === port);
if (emulatorAlreadyBooted) { // ...
```
Just to add clarity of how what this condition means to the casual glance of the maintainer.